pkg/observability/tracker.go

```diff
@@ -0,0 +1,178 @@
+package observability
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"runtime"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracehttp"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	tracer "go.opentelemetry.io/otel/trace"
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+)
+
+// Global tracing variables
+var (
+	Tracer            tracer.Tracer
+	TracerProvider    *trace.TracerProvider
+	SkipObservability = true
```

Comment:
Could we just enable tracing var instead? I feel like this is just duplicated

Reply:
This variable is necessary for not creating spans when tracing is not required. It causes memory leaks without this variable as it's used in conditional child span creation and enabling.